tests/test_distribution.py

```diff
@@ -0,0 +1,218 @@
+import os
+import json
+import unittest
+
+try:
+    from unittest import mock
+except ImportError:
+    # `Python 2` or lower than `Python 3.3` does not
+    # have the `unittest.mock` module built-in
+    import mock
+from pythonforandroid.bootstrap import Bootstrap
+from pythonforandroid.distribution import Distribution
+from pythonforandroid.recipe import Recipe
+from pythonforandroid.util import BuildInterruptingException
+from pythonforandroid.build import Context
+
+dist_info_data = {
+    "dist_name": None,
+    "bootstrap": "sdl2",
+    "archs": ["armeabi", "armeabi-v7a", "x86", "x86_64", "arm64-v8a"],
+    "ndk_api": 21,
+    "use_setup_py": False,
+    "recipes": ["hostpython3", "python3", "sdl2", "kivy", "requests"],
+    "hostpython": "/some/fake/hostpython3",
+    "python_version": "3.7",
+}
+
+
+class TestDistribution(unittest.TestCase):
+    def setUp(self):
+        self.ctx = Context()
+        self.ctx.ndk_api = 21
+        self.ctx.android_api = 27
+        self.ctx._sdk_dir = "/opt/android/android-sdk"
+        self.ctx._ndk_dir = "/opt/android/android-ndk"
+        self.ctx.setup_dirs(os.getcwd())
+        self.ctx.recipe_build_order = [
+            "hostpython3",
+            "python3",
+            "sdl2",
+            "kivy",
+        ]
+
+    def setUp_distribution_with_bootstrap(self, bs, **kwargs):
+        # extend the setUp by configuring a distribution, because some test
+        # needs a distribution to be set to be properly tested
+        self.ctx.bootstrap = bs
+        self.ctx.bootstrap.distribution = Distribution.get_distribution(
+            self.ctx,
+            name=kwargs.pop("name", "test_prj"),
+            recipes=kwargs.pop("recipes", ["python3", "kivy"]),
+            **kwargs
+        )
+
+    def tearDown(self):
+        if self.ctx.bootstrap:
+            self.ctx.bootstrap.distribution = None
```

Comment:
Minor: just wondering if this is needed since we loose the reference to `self.ctx.bootstrap.distribution` in the line below when we do ` self.ctx.bootstrap = None`?

Reply:
Thanks!!
you are right :+1: